packages/commonwealth/client/scripts/views/components/UserTrainingSlider/UserTrainingSlider.tsx

```diff
@@ -0,0 +1,299 @@
+import { CWText } from '../component_kit/cw_text';
+import { CWButton } from '../component_kit/new_designs/CWButton';
+import './UserTrainingSlider.scss';
+import { UserTrainingCardTypes } from './types';
+
+const CARD_TYPES = {
+  [UserTrainingCardTypes.GiveUpvote]: {
+    iconURL: '/static/img/shapes/shape7.svg',
+    title: 'Give an upvote',
+    description: 'Show your support for a comment or thread by upvoting it!',
+    ctaText: 'Like a thread',
+  },
+  [UserTrainingCardTypes.CreateContent]: {
+    iconURL: '/static/img/shapes/shape8.svg',
+    title: 'Make a post or comment',
+    description: 'Share your thoughts to contribute to a community.',
+    ctaText: 'Say hello',
+  },
+  [UserTrainingCardTypes.FinishProfile]: {
+    iconURL: '/static/img/shapes/shape9.svg',
+    title: 'Finish your profile',
+    description:
+      'Check out your profile page to add any socials or additional information.',
+    ctaText: 'Complete profile',
+  },
+  [UserTrainingCardTypes.ExploreCommunities]: {
+    iconURL: '/static/img/shapes/shape10.svg',
+    title: 'Explore communities',
+    description: 'Check out other Communities on Common.',
+    ctaText: 'Explore',
+  },
+};
+
+export const UserTrainingSlider = () => {
+  const { isLoggedIn } = useUserLoggedIn();
+  const navigate = useCommonNavigate();
+  const profileId = app?.user?.addresses?.[0]?.profile?.id;
+
+  const { data: profile, isLoading: isLoadingProfile } =
+    useFetchSelfProfileQuery({
+      apiCallEnabled: isLoggedIn,
+    });
+
+  const {
+    cardTempMarkedAsCompleted,
+    setCardTempMarkedAsCompleted,
+    unsetCardTempMarkedAsCompleted,
+    clearCardsTempMarkedAsCompleted,
+    shouldHideTrainingCardsPermanently,
+    setShouldHideTrainingCardsPermanently,
+  } = useUserOnboardingSliderMutationStore();
+
+  const hideAllCards = () => {
+    setShouldHideTrainingCardsPermanently(
+      profileId,
+      UserTrainingCardTypes.GiveUpvote,
+    );
+    setShouldHideTrainingCardsPermanently(
+      profileId,
+      UserTrainingCardTypes.CreateContent,
+    );
+    setShouldHideTrainingCardsPermanently(
+      profileId,
+      UserTrainingCardTypes.FinishProfile,
+    );
+    setShouldHideTrainingCardsPermanently(
+      profileId,
+      UserTrainingCardTypes.ExploreCommunities,
+    );
+  };
+
+  const hideCardPermanently = (cardName: UserTrainingCardTypes) => {
+    setShouldHideTrainingCardsPermanently(profileId, cardName);
+    unsetCardTempMarkedAsCompleted(cardName);
+  };
+
+  const markExploreCommunitiesActionAsComplete = () => {
+    // once a user visits communities page, this action is complete
+    setCardTempMarkedAsCompleted(UserTrainingCardTypes.ExploreCommunities);
+    setShouldHideTrainingCardsPermanently(
+      profileId,
+      UserTrainingCardTypes.ExploreCommunities,
+    );
+  };
+
+  const redirectToPage = (pageName: UserTrainingCardTypes) => {
+    pageName === UserTrainingCardTypes.GiveUpvote &&
+      navigate(`/dashboard/for-you`, {}, null);
```

Comment:
this might be misleading for a user because actually you cannot upvote from this view 🤔

Reply:
I actually asked this [question](https://github.com/hicommonwealth/commonwealth/issues/7194#issuecomment-2057606132) originally from Harris and got [this](https://github.com/hicommonwealth/commonwealth/issues/7194#issuecomment-2057718701) reply. That's why we redirect there.